unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WalletHUD/WalletCardHUDController.cs

```diff
@@ -0,0 +1,127 @@
+        private void OnWalletCardVisible(bool isVisible, bool _)
+        {
+            if (ownUserProfile.isGuest)
+                return;
+
+            if (isVisible)
+            {
+                fetchEthereumManaCancellationToken = fetchEthereumManaCancellationToken.SafeRestart();
+                fetchPolygonManaCancellationToken = fetchPolygonManaCancellationToken.SafeRestart();
+                RequestEthereumManaAsync(fetchEthereumManaCancellationToken.Token).Forget();
+                RequestPolygonManaAsync(fetchPolygonManaCancellationToken.Token).Forget();
+            }
+            else
+            {
+                fetchEthereumManaCancellationToken.SafeCancelAndDispose();
+                fetchPolygonManaCancellationToken.SafeCancelAndDispose();
+            }
+        }
+
+        private async UniTask RequestEthereumManaAsync(CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                await UniTask.WaitUntil(() =>
+                    ownUserProfile != null &&
+                    !string.IsNullOrEmpty(ownUserProfile.userId) &&
+                    !dataStore.wallet.isWalletSectionVisible.Get(),
+                    cancellationToken: cancellationToken);
+
+                view.SetEthereumManaLoadingActive(true);
+                Promise<double> promise = theGraph.QueryEthereumMana(ownUserProfile.userId);
+                if (promise != null)
+                {
+                    await promise;
+                    dataStore.wallet.currentEthereumManaBalance.Set(promise.value);
+                    view.SetEthereumManaLoadingActive(false);
+                }
+
+                await UniTask.Delay(TimeSpan.FromSeconds(WalletUtils.FETCH_MANA_INTERVAL), cancellationToken: cancellationToken);
+
+                if (cancellationToken.IsCancellationRequested)
+                    break;
+            }
+        }
+
+        private async UniTask RequestPolygonManaAsync(CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                await UniTask.WaitUntil(() =>
+                    ownUserProfile != null &&
+                    !string.IsNullOrEmpty(ownUserProfile.userId) &&
+                    !dataStore.wallet.isWalletSectionVisible.Get(),
+                    cancellationToken: cancellationToken);
+
+                view.SetPolygonManaLoadingActive(true);
+                Promise<double> promise = theGraph.QueryPolygonMana(ownUserProfile.userId);
+                if (promise != null)
+                {
+                    await promise;
+                    dataStore.wallet.currentPolygonManaBalance.Set(promise.value);
+                    view.SetPolygonManaLoadingActive(false);
+                }
+
+                await UniTask.Delay(TimeSpan.FromSeconds(WalletUtils.FETCH_MANA_INTERVAL), cancellationToken: cancellationToken);
+
+                if (cancellationToken.IsCancellationRequested)
+                    break;
+            }
```

Comment:
Can we merge these methods? they have the same structure, they wait for the same conditions and they wait the same time, but make different requests, we can do both in just one async loop

Reply:
You're completely right, we can merge it into only 1 method, even in the TheGraph service side.

Done here: https://github.com/decentraland/unity-renderer/pull/5318/commits/7f73546978d48378fa8d36332dddd885670ce071